src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketGitSCMBuilder.java

```diff
@@ -170,22 +170,14 @@ public BitbucketGitSCMBuilder withBitbucketRemote() {
                         credentialsId(),
                         StandardCredentials.class
                 );
-        Integer protocolPortOverride = null;
         BitbucketRepositoryProtocol protocol = credentials instanceof SSHUserPrivateKey
                 ? BitbucketRepositoryProtocol.SSH
                 : BitbucketRepositoryProtocol.HTTP;
+        String cloneLink = null;
         if (protocol == BitbucketRepositoryProtocol.SSH) {
```

Comment:
As per @casz suggestion, I would change this block to something like:
```
for (BitbucketHref link : cloneLinks()) {
    if (protocol.getType().equals(link.getName())) {
        cloneLink = link.getHref();
        break;
    }
}
```
Since [protocol.getType()](https://github.com/jenkinsci/bitbucket-branch-source-plugin/blob/master/src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketRepositoryProtocol.java) already matches the link names.

Reply:
This part is done.